api/src/main/java/org/openmrs/module/ModuleUtil.java

```diff
@@ -571,7 +577,15 @@ public static void expandJar(File fileToExpand, File tmpModuleDir, String name,
 					String entryName = jarEntry.getName();
 					// trim out the name path from the name of the new file
 					if (!keepFullPath && name != null) {
-						entryName = entryName.replaceFirst(name, "");
+						if (name.equals(entryName)) {
+							int lastSlash = name.lastIndexOf('/');
+							if (lastSlash >= 0) {
+								entryName = entryName.replaceFirst(name.substring(0, lastSlash + 1), "");
```

Comment:
@wluyima what about extracting it into a private method with a descriptive name instead of a comment? A lot of methods are already bloated and hard to read since they do so many things